pkg/endpoint/bpf.go

```diff
@@ -448,6 +448,8 @@ func (e *Endpoint) regenerateBPF(owner Owner, epdir string, regenContext *Regene
 	if e.SecurityIdentity != nil {
 		stats.policyCalculation.Start()
 		err = e.regeneratePolicy(owner)
+		// Set the final policyStatus for the endpoint.
+		stats.policyStatus = e.getPolicyStatus()
```

Comment:
Let's remove this and check the build status. It would be misleading to report an endpoint with policy enforcement enabled if the build for the endpoint fails afterward. Same for an existing endpoint that is being rebuilt, the policy enforcement status metric should stay the same until the endpoint has been successfully built as it is only then that a potentially new policy enforcement status becomes effective.

Reply:
@tgraf I'm not sure if your reply works at all. I need to set up the `stats.policyStatus` in any place, to send the value to sendMetrics. 

- If I do in the defer function, I need to lock in the endpoint (Something rejected before) 
- If I send the endpoint in the sendMetrics I'll do the lock in SendMetrics.

I'm not sure If I get your point  :\

Reply:
I did not reject to lock the endpoint. I stated that the endpoint was unlocked when calling `getPolicyStatus()`.